From 308d2cc2befeb3adf1aeef4729fecf27bccc79bb Mon Sep 17 00:00:00 2001
From: Jan Altensen <info@stricted.net>
Date: Wed, 9 Jan 2019 22:58:01 +0100
Subject: [PATCH 4/4] build: only write build_number.txt when BUILD_NUMBER has
 changed

* prevents $(file <.../out/build_number.txt) was changed, regenerating...

Change-Id: I801be27695fd983bee01d0b0d4ef0948392cb849
---
 core/main.mk | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/core/main.mk b/core/main.mk
index 97ec295f8..93411c965 100644
--- a/core/main.mk
+++ b/core/main.mk
@@ -63,9 +63,11 @@ include $(BUILD_SYSTEM)/clang/config.mk
 # Write the build number to a file so it can be read back in
 # without changing the command line every time.  Avoids rebuilds
 # when using ninja.
-$(shell mkdir -p $(OUT_DIR) && \
-    echo -n $(BUILD_NUMBER) > $(OUT_DIR)/build_number.txt)
 BUILD_NUMBER_FILE := $(OUT_DIR)/build_number.txt
+ifneq ($(shell cat $(BUILD_NUMBER_FILE)),$(BUILD_NUMBER))
+$(shell mkdir -p $(OUT_DIR) && \
+    echo -n $(BUILD_NUMBER) > $(BUILD_NUMBER_FILE))
+endif
 
 ifeq ($(HOST_OS),darwin)
 DATE_FROM_FILE := date -r $(BUILD_DATETIME_FROM_FILE)
-- 
2.17.1

